src/Pact/Eval.hs

```diff
@@ -458,22 +565,30 @@ evaluateDefs info mdef defs = do
       evalError i $ "Recursion detected: " <> prettyList pl
 
   -- the order of evaluation matters for 'dresolve' - this *must* be a left fold
-  let dresolve ds (d,dn,_) = HM.insert dn (Ref $ unify ds <$> d) ds
-      unifiedDefs = foldl' dresolve HM.empty sortedDefs
-
-  traverse (runSysOnly . evalConsts) unifiedDefs
+  -- Todo: Charge gas here.
+  isExecutionFlagSet FlagDisableInlineMemCheck >>= \case
```

Comment:
this syntax is weird but I get how it's nice, but consider instead `ifExecutionFlagSet`

Reply:
`ifExecutionFlagSet` is a bit inconvenient since I need to do two different blocks.